src/main/java/org/reactome/release/orthopairs/UniProtGeneNamesRetriever.java

```diff
@@ -0,0 +1,178 @@
+     */
+    public static List<Set<String>> partitionUniProtIdentifiers(Set<String> uniprotIds) {
+        List<Set<String>> partitionedUniProtIds = new ArrayList<>();
+        Set<String> partition = new HashSet<>();
+        for (String uniprotId : uniprotIds) {
+            partition.add(uniprotId);
+            // Once the partition has 250 identifiers, add it to the List and reset the partition variable.
+            if (partition.size() == MAX_UNIPROT_BATCH_QUERY_SIZE) {
+                partitionedUniProtIds.add(partition);
+                partition = new HashSet<>();
+            }
+        }
+        // The final identifiers at the end are added to the List here.
+        partitionedUniProtIds.add(partition);
+        return partitionedUniProtIds;
+    }
+
+    /**
+     * Query UniProt via the Java API for gene names using UniProt accession IDs.
+     * @param partitionedUniProtIds List<Set<String>> Partitioned UniProt identifiers.
+     * @return Set<String> - Accession-to-Name mappings that will be stored to a file.
+     * @throws ServiceException - Thrown by UniProtService class if service is unavailable.
+     * @throws InterruptedException - Thrown if the Sleep process that occurs after every batch query is interrupted.
+     */
+    public static Set<String> retrieveGeneNamesFromUniProt(List<Set<String>> partitionedUniProtIds) throws ServiceException, InterruptedException {
+
+        // Create the UniProtService.
+        ServiceFactory serviceFactoryInstance = Client.getServiceFactoryInstance();
+        UniProtService uniprotService = serviceFactoryInstance.getUniProtQueryService();
+        uniprotService.start();
+        int count = 0;
+        Set<String> uniprotAccessionsToGeneNames = new HashSet<>();
+        for (Set<String> uniprotIdentifierPartition : partitionedUniProtIds) {
+            // Build UniProt API query from Set of 250 UniProt identifiers.
+            Query query = UniProtQueryBuilder.accessions(uniprotIdentifierPartition);
+            // Perform UniProt API query to retrieve gene names associated with identifiers.
+            QueryResult<UniProtComponent<Gene>> uniprotEntries = uniprotService.getGenes(query);
+
+            while (uniprotEntries.hasNext()) {
+                count++;
+                // Get Gene object returned from UniProt.
+                UniProtComponent<Gene> geneObject = uniprotEntries.next();
+                if (!geneObject.getComponent().isEmpty()) {
+                    // Iterate through all Gene components in the response.
+                    for (Gene geneComponent : geneObject.getComponent()) {
+                        // Tab-separate UniProt accession ID and its associated gene name, and then store these in the Set that will be returned.
+                        uniprotAccessionsToGeneNames.add(geneObject.getAccession().toString() + "\t" + geneComponent.getGeneName().toString() + "\n");
+                    }
+                }
```

Comment:
I think the `if` statement can be removed since the for loop will not be entered if `geneObject.getComponent()` is empty.